src/Bicep.Core/TypeSystem/Az/OciAzTypeLoader.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.IO;
+using Azure.Bicep.Types;
+using System.Formats.Tar;
+using System.IO.Compression;
+using System;
+using System.Collections.Immutable;
+
+namespace Bicep.Core.TypeSystem.Az
+{
+    public class OciAzTypeLoader : TypeLoader
+    {
+        private readonly ImmutableDictionary<string, byte[]> typesCache;
+
+        public OciAzTypeLoader(ImmutableDictionary<string, byte[]> typesCache)
+        {
+            this.typesCache = typesCache;
+        }
+
+        public static OciAzTypeLoader FromTgz(string pathToGzip)
+        {
+            var typesCache = ImmutableDictionary.CreateBuilder<string, byte[]>();
+            using var fileStream = File.OpenRead(pathToGzip);
+            using var gzipStream = new GZipStream(fileStream, CompressionMode.Decompress);
+            using var tarReader = new TarReader(gzipStream, leaveOpen: true);
+            while (tarReader.GetNextEntry(copyData: true) is { } entry)
```

Comment:
Why copyData?  You're already copying from a stream into a new MemoryStream.  And then into bytes.  In fact, can you not use entry.DataStream.ReadBytes() and skip all the extra copying of data?

Reply:
I tried but I get this:
![image](https://github.com/Azure/bicep/assets/9611108/0ec551f2-25ea-49dc-a043-54d05325818d)